src/components/ReadyForPrint.tsx

```diff
@@ -66,7 +73,7 @@ async function waitForDataLoading(dataLoadingIsDone: DataLoading['isDone']) {
   let done: boolean = dataLoadingIsDone();
 
   while (!done) {
-    await new Promise((r) => setTimeout(r, 1000));
+    await new Promise((resolve) => window.requestIdleCallback(resolve));
```

Comment:
TIL, very cool 🤯 

Reading the docs on this, there is a note

> Note: A timeout option is strongly recommended for required work, as otherwise it's possible multiple seconds will elapse before the callback is fired.

Is this a problem? Seeing as we currently block the users process/next until PDF is finished. I.e. _how_ unfair does scheduling become by using this

Reply:
Good catch, the reason I changed this was i thought waiting 1 second for each check was a bit much, i'll add a timeout 👍